test/compilable/test17419.d

```diff
@@ -39,11 +39,13 @@ void bar()
 class FooD {}
 interface FooDInterface {}
 extern (C++) class FooCpp {}
+extern (C++) struct FooCppStruct {}
 extern (C++) interface FooCppInterface {}
 
 static assert(__traits(getLinkage, FooD) == "D");
 static assert(__traits(getLinkage, FooDInterface) == "D");
 static assert(__traits(getLinkage, FooCpp) == "C++");
+static assert(__traits(getLinkage, FooCppStruct) == "C++");
```

Comment:
When this now works for struct, the documentation should be updated [1].

[1] https://dlang.org/spec/traits.html#getLinkage